.github/workflows/codeql.yml

```diff
@@ -0,0 +1,57 @@
+name: CodeQL
+
+on:
+  push:
+    branches:
+      - main
+      - release/v*
+      - feat/ci/codeql-and-scorecard
+  pull_request:
+    branches:
+      - main
+      - release/v*
+      - feat/ci/codeql-and-scorecard
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: ["go"]
```

Comment:
Will we add other languages in a separate PR?

Reply:
Which languages do we want to support?

Reply:
All the languages! xD Which are available? Not sure, bash, python, ..?

Reply:
Supported languages: https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/

Only Python is used in our project. Since CodeQL is already running ~10min, I would refrain from adding Python, because the value would be marginal.

Reply:
Okay, thanks for checking. :) Would also be ok for me to only add Go right now, just wanted to know if you're planning to extend this.